### PR TITLE
feat(anix): Add new provider anix.to

### PR DESCRIPTION
**Pull request by hase0278**
**What kind of change does this PR introduce?**
New provider. The new provider provides alternative source for those who wants japanese anime and chinese anime(donghua) in their site. Link to provider:
https://anix.sh

**Did you add tests for your changes?**
Yes I added tests

**If relevant, did you update the documentation?**
No, not yet

**Summary**
Motivation for the new provider is gogoanime not updating their recent episodes recently.

### CODE REVIEW

**Comment by hase0278** [comment]
I won't add more commit to this pr for now unless asked after review

File: src/models/types.ts [hunk 1]
```diff
@@ -177,6 +180,7 @@ export enum StreamingServers {
   MyCloud = 'mycloud',
   Filemoon = 'filemoon',
   VidStreaming = 'vidstreaming',
+  VidStream = 'vidstream',
```
[the remark this answers is not in the document]
Comment (hase0278) [reply src/models/types.ts:183]: I think it is different. It is not embedded but a default player for anix I think that utilizes JW player. You can see here when you inspect element:
https://anix.sh/anime/ao-no-miburo/ep-1

[the remark this answers is not in the document]
Comment (hase0278) [reply src/models/types.ts:183]: Look for loadIframePlayer using ctrl+f thats why I came to the conclusion that it is different.

File: src/providers/anime/anix.ts [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+  private readonly requestedWith = 'XMLHttpRequest';
+
+  constructor(customBaseURL?: string, proxy?: ProxyConfig, adapter?: AxiosAdapter) {
+    super(...arguments);
+    this.baseUrl = customBaseURL
+      ? customBaseURL.startsWith('http://') || customBaseURL.startsWith('https://')
+        ? customBaseURL
+        : `http://${customBaseURL}`
+      : this.baseUrl;
+    if (proxy) {
```
[the remark this answers is not in the document]
Comment (hase0278) [reply src/providers/anime/anix.ts:34]: Ok will do. Commit will come later I am at work now.

[the remark this answers is not in the document]
Comment (hase0278) [reply src/providers/anime/anix.ts]: Done. Please review again.

**Review by riimuru — APPROVED** [review]
@hase0278 Thank you for the changes, and sorry for the late review. Yeah adding `Builtin` now makes more sense, Thanks.

**Pull request merged**